src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,146 @@
+            case 'DIGITEVEN':
+                return array_of_digits.filter(digit => +digit % 2 === 0);
+            default:
+                throw new Error('Cannot Determine Winning numbers.');
+        }
+    };
+
+    const highlightWinningNumbers = (winning_numbers) => {
+        winning_numbers.forEach(digit => {
+            const element = el_container.querySelector(`.digit-${digit}`);
+            element.classList.remove('digit-losing');
+            element.classList.add('digit-winning');
+        });
+    };
+
+    const observeResize = () => {
+        window.onresize = () => {
+            if (el_peek_box) {
+                adjustBoxSizes();
+                el_peek_box.setAttribute('style', `transform: translateX(${calculateOffset()}px)`);
+            }
+        };
+    };
+
+    // Calculate left margin of the peek-box against the "width" of the container
+    const calculateLeftMargin = (width) => (width - (digit_block_size * 10)) / 2;
+
+    // Calculate peek-box left offset.
+    const calculateOffset = () => {
+        const left_offset = calculateLeftMargin(el_container.offsetWidth);
+        return (+current_spot * digit_block_size) + left_offset - style_offset_correction;
+    };
+
+    const markAsLost = () => {
+        if (!el_peek_box || !el_peek) {
+            setElements();
+        }
+        el_peek.classList.remove('digit-winning', 'digit-running');
+        el_peek_box.classList.remove('digit-winning', 'digit-running');
+        el_peek.classList.add('digit-losing');
+        el_peek_box.classList.add('digit-losing');
+    };
+
+    const markAsWon = () => {
+        if (!el_peek_box || !el_peek) {
+            setElements();
+        }
+        el_peek_box.classList.remove('digit-losing', 'digit-running');
+        el_peek_box.classList.add('digit-winning');
+        el_peek.classList.remove('digit-losing', 'digit-running');
+        el_peek.classList.add('digit-winning');
+    };
+
+    const setElements = () => {
+        el_peek     = el_container.querySelector('.peek');
+        el_peek_box = el_container.querySelector('.peek-box');
+        el_mask     = el_peek_box.querySelector('.peek-box > .mask');
+    };
+
+    const update = (current_tick_count, { quote }) => {
+        setElements();
+        el_container.classList.remove('invisible');
+        adjustBoxSizes();
+        current_spot = quote.substr(-1);
+
+        el_mask.innerText = `${current_tick_count}/${total_tick_count}`;
+        el_peek.innerText = current_spot;
+
+        el_peek_box.classList.add('digit-running');
+        el_peek.classList.add('digit-running');
+
+        el_peek_box.setAttribute('style', `transform: translateX(${calculateOffset()}px)`);
+
+        if (contract_status === 'won') {
+            markAsWon();
+        }
+        if (contract_status === 'lost') {
+            markAsLost();
+        }
+    };
+
+    const remove = () => {
+        window.onresize = null;
```

Comment:
do we want to only remove that specific function? what if there are other resize functions being set from others that we want to keep?

Reply:
there is always 1 resize function. they don't stack.